query.py

```diff
@@ -0,0 +1,394 @@
+
+        resources_to_add = set(resources_to_add)
+
+        if not self.main_internal_name:
+            # if there was no where condition, default selection of the
+            # main resource
+            self.main_internal_name = resources_to_add[0]
+
+        # loop to include information about resources other than the
+        # main resource
+        for internal_name in resources_to_add:
+            if internal_name != self.main_internal_name:
+                url_temp = ""
+                internal_path = nx.shortest_path(
+                    self.rscs_graph,
+                    source=self.main_internal_name,
+                    target=internal_name,
+                )
+                for ind in range(len(internal_path) - 1):
+                    edge = self.rscs_graph.edges[
+                        internal_path[ind], internal_path[ind + 1]
+                    ]
+                    if "include" in edge:
+                        if url_temp:
+                            url_temp += (
+                                "&"
+                                + "_include:iterate="
+                                + edge["include"]
+                            )
+                        else:
+                            url_temp = "_include=" + edge["include"]
+                    elif "revinclude" in edge:
+                        if url_temp:
+                            url_temp += (
+                                "&"
+                                + "_revinclude:iterate="
+                                + edge["revinclude"]
+                            )
+                        else:
+                            url_temp = (
+                                "_revinclude=" + edge["revinclude"]
+                            )
+                if self.url_rev_include:
+                    self.url_rev_include += "&" + url_temp
+                else:
+                    self.url_rev_include += url_temp
+
+    def _get_rev_include_possibilitie(
+        self, CapabilityStatement_path: str = None
+    ):
+        """Builds a dictionary that will indicate for each type of 
+        resource which are its mother resources (revinclude) and its 
+        daughter resources (include).
+
+        Arguments:
+            CapabilityStatement_path {str} -- path to the json file that
+            contains a resource of type CapabilityStatement
+        """
+        if CapabilityStatement_path:
+            CapabilityStatement = self._get_CapabilityStatement_from_file(
+                CapabilityStatement_path
+            )
+        else:
+            CapabilityStatement = (
+                self._get_CapabilityStatement_from_api()
+            )
+
+        dict_reference = dict()
+        for ressource in CapabilityStatement["resource"]["rest"][0][
+            "resource"
+        ]:  # check the 0 , we could have several
+            type = ressource["type"]
+            dict_reference[type] = dict()
+            try:
```

Comment:
Why a try/except?
I'm sure you can use an if instead

Reply:
Done :)